data-raw/globalearthquake.R

```diff
@@ -0,0 +1,45 @@
+library(readr)
+library(vroom)
+library(dplyr)
+library(data.table)
+
+
+csv_file <- file.path("data-raw", "worldwide_seisme.csv")
+if (!file.exists(csv_file)) {
+  deb <- c(2010, 2000, 1990, 1980, 1970, 1960, 1950, 1940, 1930, 1920, 1910, 1900)
+  fin <- c(2020, 2010, 2000, 1990, 1980, 1970, 1960, 1950, 1940, 1930, 1920, 1910)
+
+  url <- paste("https://earthquake.usgs.gov/fdsnws/event/1/query?format=csv&starttime=", as.character(deb[1]), "-01-01&endtime=", as.character(fin[1]), "-01-01&minmagnitude=5", sep = "")
+  combined_data <- read_csv(url)
+
+  for (i in 2:12) {
+    url <- paste("https://earthquake.usgs.gov/fdsnws/event/1/query?format=csv&starttime=", as.character(deb[i]), "-01-01&endtime=", as.character(fin[i]), "-01-01&minmagnitude=5", sep = "")
+    newz <- read_csv(url)
+    combined_data <- bind_rows(combined_data, newz)
```

Comment:
On way to optimize this is to use the fact that `bind_rows` can take a list of data frames as input. So you can do
`combined_data <- list()` and in the loop
 `combined_data <- c( combined_data, list(newz))`
and then after the loop
 `combined_data <- bind_rows(combined_data)`

Reply:
Or even better, let vroom do that itself ;-)